sqlmesh/core/renderer.py

```diff
@@ -269,20 +262,9 @@ def time_column_filter(self, start: TimeLike, end: TimeLike) -> exp.Between:
             self._time_converter(start), self._time_converter(end)  # type: ignore
         )
 
-    @property
-    def contains_star_query(self) -> bool:
-        """Returns True if the model's query contains a star projection."""
-        return any(isinstance(expression, exp.Star) for expression in self.render().expressions)
-
     def update_schema(self, schema: MappingSchema) -> None:
-        old_schema = self._schema
         self._schema = schema
-
-        if self.contains_star_query and old_schema != schema:
-            new_cache = {}
-            for cached_key, cached_query in self._query_cache.items():
-                new_cache[cached_key] = self._optimize_query(cached_query) or cached_query
-            self._query_cache = new_cache
+        self._query_cache.clear()
```

Comment:
This kills the benefit of having a cache in the first place, doesn't it?

Reply:
Basically we'll never use cached rendered query since all of them will be immediately wiped out once we set the schema.

Reply:
The schema is set once though, no? I think that for any subsequent `render_query` we should be hitting the cache?

Reply:
It's set after we load from the cache though.

Reply:
I'm talking about the file system cache, not the in-memory one.

Reply:
Ah, I thought you meant the latter (`_query_cache`), so I'm probably missing context regarding that. Thanks for clarifying.

Reply:
@izeigerman well then isn't the current cache wrong?